helm-charts/chatqna/templates/horizontalPorAutoscaler.yaml

```
@@ -0,0 +1,50 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+{{- if .Values.horizontalPodAutoscaler.enabled }}
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ include "chatqna.fullname" . }}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{ include "chatqna.fullname" . }}
+  minReplicas: 1
+  maxReplicas: 6
+  metrics:
+  - type: Object
+    object:
+      metric:
+        # TGI time metrics are in seconds
+        name: tgi_request_latency
+      describedObject:
+        apiVersion: v1
+        # get metric for named object of given type (in same namespace)
+        kind: Service
+        name: tgi-svc
+      target:
+        # tgi_request_latency is already average for all the TGI pods,
+        # so this uses Value instead of averageValue.
+        # On ICL Xeon, max TGI queue wait time target = 4s.
```

Let's add link to docs in case user wants to experiment with this (also to other HPA files):
```suggestion
        # tgi_request_latency is average for all the TGI pods. To avoid replica fluctuations when
        # TGI startup + request processing takes longer than HPA evalution period, this uses
        # "Value" (replicas = metric.value / target.value), instead of "averageValue" type:
        #  https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#algorithm-details
```